src/rgw/rgw_rest_s3.cc

```diff
@@ -4500,6 +4500,8 @@ AWSGeneralAbstractor::get_auth_data_v4(const req_state* const s,
     throw ret;
   }
 
+  s->auth.s3_postobj_creds.access_key = access_key_id;
+
```

Comment:
this doesn't compile because these functions only have const access to req_state

Reply:
Yes you are right. I made a mistake but from function `AWSEngine::authenticate` `req_state` is const and so we can't get access_key out via `req_state`.
Any suggestion for this?